platforms/jvm/language-java/src/main/java/org/gradle/api/internal/tasks/compile/CommandLineJavaCompiler.java

```diff
@@ -57,11 +58,22 @@ public WorkResult execute(JavaCompileSpec spec) {
     }
 
     private ExecHandle createCompilerHandle(String executable, JavaCompileSpec spec) {
-        ExecHandleBuilder builder = execHandleFactory.newExec();
+        ClientExecHandleBuilder builder = execHandleFactory.newExec();
         builder.setWorkingDir(spec.getWorkingDir());
         builder.setExecutable(executable);
-        argumentsGenerator.collectArguments(spec, new ExecSpecBackedArgCollector(builder));
-        builder.setIgnoreExitValue(true);
+        argumentsGenerator.collectArguments(spec, new ArgCollector() {
```

Comment:
💭 Here we just inlined `ExecSpecBackedArgCollector` and removed `ExecSpecBackedArgCollector.java`